### ISSUE TITLE
Getting an exception when updating http listener with an existing waf-policy

### ISSUE THREAD

**Issue by timvanderweijde** [Azure/azure-cli#15592]
## Describe the bug
When updating an http listener with an existing waf-policy, the command returns an exception

**Command Name**
`az network application-gateway http-listener update`

**Errors:**
```
BadRequestError: Cannot parse the request.
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- `a Waf Policy must be created. The name must be used in the next command`
- `az network application-gateway http-listener update -g {} --gateway-name {} -n {} --waf-policy {}`

## Expected Behavior
Waf-policy is connected to the http listener without any errors.

## Environment Summary
```
Windows-10-10.0.19041-SP0
Python 3.6.8
Installer: MSI

azure-cli 2.13.0

```

**Comment by timvanderweijde** [comment]
Is there any other way to associate the waf-policy with the http listener? (in the portal you can associate the http listener when browing the waf-policy itself, is there a az cli command for this?)

**Comment by yonzhan** [comment]
add to S178

**Comment by timvanderweijde** [comment]
This is also not working with the `create` command.

This is the debug part of the exception.

msrest.http_logger : {
  "error": {
    "code": "InvalidRequestFormat",
    "message": "Cannot parse the request.",
    "details": [
      {
        "code": "InvalidJsonReferenceFormat",
        "message": "Reference Id /subscriptions/<some-guid>/resourceGroups/<some-resourcegroup>r/providers/Microsoft.Network/applicationGateways/<some-ag-name>/firewallPolicy/ag-waf-policy-listeners is not formatted correctly. The Id is expected to reference resources of type ApplicationGatewayWebApplicationFirewallPolicies. Path properties.httpListeners[1].properties.firewallPolicy."
      }
    ]
  }
}

**Comment by timvanderweijde** [comment]
Somehow using the `name' of the waf-policy doesn't work. (The documentation is saying you can use the `name`)
Retrieving the ID and using the full ID works though.

**Comment by haroldrandom** [comment]
@timvanderweijde It's a bug, I will plan to fix it ASAP

**Comment by haroldrandom** [comment]
@timvanderweijde The patch is merged and will be shipped with Azure CLI 2.16.0, you could have a try after 12/8/2020. Or you could try an edge build from https://dev.azure.com/azure-sdk/public/_build/results?buildId=620599&view=artifacts&type=publishedArtifacts
